test/e2e/node_maintenance_test.go

```diff
@@ -256,6 +256,42 @@ var _ = Describe("Starting Maintenance", func() {
 		})
 
 	})
+	Context("for a worker node with a low eviction timeout", func() {
+
+		var maintenanceNodeName string
+		var nodeMaintenance *nmo.NodeMaintenance
+
+		BeforeEach(func() {
+			createTestDeployment("test-deployment-2")
+			maintenanceNodeName = getTestDeploymentNodeName()
+			nodeMaintenance = getNodeMaintenance(testMaintenance, maintenanceNodeName, time.Second*10)
+			Expect(createCRIgnoreUnrelatedErrors(nodeMaintenance)).To(Succeed())
+			DeferCleanup(Client.Delete, context.TODO(), nodeMaintenance)
+		})
+
+		When("evction timeout is low", func() {
+			It("should succeed and be reconciled with fixed duration at least once", func() {
+				By("verify CR has phase is succeeded")
+				Eventually(func() (bool, error) {
+					nm := &nmo.NodeMaintenance{}
+					if err := Client.Get(context.TODO(), types.NamespacedName{Name: nodeMaintenance.Name}, nm); err != nil {
+						return false, err
+					}
+
+					if nm.Status.Phase != nmo.MaintenanceSucceeded {
+						logInfof("phase: %s\n", nm.Status.Phase)
+						return false, nil
+					}
+
+					return true, nil
+				}, 300*time.Second, 10*time.Second).Should(BeTrue(), "maintenance did not succeed in time")
+				By("verify printed log")
+				// check operator log showing it reconciled with fixed duration because of drain timeout
+				// it should be caused by the test deployment's termination graceperiod > drain timeout
+				Expect(getOperatorLogs()).To(ContainSubstring(nodemaintenance.FixedDurationReconcileLog))
```

Comment:
>comment doesn't match because you removed the termination graceperiod

There is a [default value](https://github.com/kubernetes/api/blob/v0.29.2/core/v1/types.go#L3117) of 30 seconds for the graceperiod when it isn't set